app/renderer/abr-document.js

```diff
@@ -319,23 +332,26 @@ AbrDocument.prototype = {
         if (!path) {
             return this.saveAs(callback);
         }
+        // Pause the watcher to avoid triggering callbacks after saving the document
+        this.pauseWatcher();
         var that = this,
-            data = this.getData(),
-            callback2 = function (err) {
-                if (err) {
-                    return dialogs.fileAccessDenied(path, function () {
-                        that.saveAs(callback);
-                    });
-                }
-                that.setClean();
-                that.setPath(path);
-                that.ipcClient.trigger("setWinPath", path);
-                that.updateWindowTitle();
-                if (typeof callback === "function") {
-                    callback();
-                }
-            };
-        return files.writeFile(data, path, callback2);
+            data = this.getData();
+        files.writeFile(data, path, function (err) {
+            if (err) {
+                return dialogs.fileAccessDenied(path, function () {
+                    that.saveAs(callback);
+                });
+            }
+            that.setClean();
+            that.setPath(path);
+            that.ipcClient.trigger("setWinPath", path);
+            that.updateWindowTitle();
+            if (typeof callback === "function") {
+                callback();
+            }
+            that.startWatcher();
+        });
+        return true;
```

Comment:
I only added it to stay consistent with `saveAs()`.
Since it can return `false` if it gets an invalid path, I thought it should return `true` if the function doesn't exit early.

Should we get rid of these boolean returns altogether?